File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):
    cards1_idx = 0
    cards2_idx = 0
    res = True
    
    for word in goal:
        if(cards1_idx < len(cards1) and cards1[cards1_idx] == word):
            cards1_idx += 1
            continue
        elif(cards2_idx < len(cards2) and cards2[cards2_idx] == word):
            cards2_idx += 1
            continue
        else:
            res = False
            break
    
    if(res == True):
        return 'Yes'
    else:
        return 'No'
            

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.js
function solution(cards1, cards2, goal) {
    let answer = 'Yes';
    
    for(let i = 0 ; i < goal.length; i++){
        if(cards1[0] === goal[i]){
            cards1.shift();
            continue;
        }else if(cards2[0] === goal[i]){
            cards2.shift();
            continue;
        }else{
            answer = 'No';
            break;
        }
    }
    return answer;
}